protos/file2.proto

```diff
@@ -0,0 +1,206 @@
+//
+// Data buffers can be placed in three different places:
+//
+// * Global data buffers are placed in the file footer.  These should contain
+//   data that is common across the file (such as the file schema or a common
+//   dictionary)
+// * Column metadata buffers are placed in the column metadata block.  These
+//   should contain data that is common across all pages within the column.
+// * Page data buffers are written in the data pages section of the file.  It
+//   is expected that these will contain the vast majority of the file's data.
+//
+// ## File Layout
+//
+// Note: the number of pages (PN) is independent of the number of columns (CN)
+//       and each page/column can have any number of buffers (Px_N)/(Cx_N)
+//       which is determined by the encodings.
+//
+//       Buffers often need to be aligned.  64-byte alignment is common when
+//       working with SIMD operations.  4096-byte alignment is common when
+//       working with direct I/O.  In order to ensure these buffers are aligned
+//       writers may need to insert padding before the buffers.
+//       
+//       If direct I/O is required then most (but not all) fields described
+//       below must be sector aligned.  We have marked these fields with an
+//       asterick for clarity.  Readers should assume there will be optional
+//       padding inserted before these fields.
+//
+//       All footer fields are unsigned integers written with  little endian
+//       byte order.
+//
+// ├──────────────────────────────────┤
+// | Data Pages                       |
+// |   Page 0, Buffer 0*              |
+// |   ...                            |
+// |   Page 0, Buffer P0_N*           |
+// |   Page 1, Buffer 0*              |
+// |   ...                            |
+// |   Page PN, Buffer PN_N*          |
+// ├──────────────────────────────────┤
+// | Column Metadatas                 |
+// | |A| Column 0 Metadata*           |
+// |     Column 0 Meta Buffer 0*      |
+// |     ...                          |
+// |     Column 0 Meta Buffer C0_N*   |
+// |     Column 1 Metadata*           |
+// |     ...                          |
+// |     Column CN Meta Buffer CN_N*  |
+// ├──────────────────────────────────┤
+// | Column Metadata Offset Table     |
+// | |B| Column 0 Metadata Position*  |
+// |     Column 0 Metadata Size       |
+// |     ...                          |
+// |     Column CN Metadata Position  |
+// |     Column CN Metadata Size      |
+// ├──────────────────────────────────┤
+// | Global Buffers                   |
+// | |C| Global Meta Buffer 0*        |
+// |     ...                          |
+// |     Global Meta Buffer GN*       |
+// ├──────────────────────────────────┤
+// | Global Buffers Offset Table      |
+// | |D| Global Buffer 0 Position*    |
+// |     Global Buffer 0 Size         |
+// |     ...                          |
+// |     Global Buffer GN Position    |
+// |     Global Buffer GN Size        |
```

Comment:
My initial thought here is that it's unclear why this is preferable to just having a protobuf message for global data. Protobuf (or some other serialization framework) gives us an easy way to describe messages and handling changes to the message. I worry we are taking on some engineering overhead to add new types of metadata to the format.